src/Matcher.js

```diff
@@ -39,6 +39,16 @@ export default class Matcher {
     });
   }
 
+  joinPath(basePath, path) {
+    if (!path) {
+      return basePath;
+    }
+    if (basePath.charAt(basePath.length - 1) === '/') {
+      return `${basePath}${path}`;
+    }
+    return `${basePath}/${path}`;
```

Comment:
might need to canonicalize `basePath` as well in case it's missing the leading slash... maybe not